internal/controller/postgrescluster/util_test.go

```diff
@@ -325,42 +341,44 @@ func TestAddNSSWrapper(t *testing.T) {
 
 			addNSSWrapper(image, imagePullPolicy, template)
 
-			// verify proper nss_wrapper env vars
-			var expectedContainerUpdateCount int
-			for i, c := range template.Spec.Containers {
-				if c.Name == "database" || c.Name == "pgbackrest" {
-					assert.DeepEqual(t, expectedEnv, c.Env)
-					expectedContainerUpdateCount++
-				} else {
-					assert.DeepEqual(t, beforeAddNSS[i], c)
+			t.Run("container-updated", func(t *testing.T) {
+				// Each container that requires the nss_wrapper envs should be updated
+				var actualUpdatedContainerCount int
+				for i, c := range template.Spec.Containers {
+					if c.Name == naming.ContainerDatabase ||
+						c.Name == naming.PGBackRestRepoContainerName ||
+						c.Name == naming.ContainerPGUpgrade ||
+						c.Name == naming.PGBackRestRestoreContainerName {
+						assert.DeepEqual(t, expectedEnv, c.Env)
+						actualUpdatedContainerCount++
+					} else {
+						assert.DeepEqual(t, beforeAddNSS[i], c)
```

Comment:
❓ I think the contents of `beforeAddNSS` are modified along with `template.Spec.Containers`. Does this assertion pass if it is moved outside of the `else` branch? If so, use `DeepCopy()` around L340 to ensure that "before" values stay separate.